From 7a9e7670315172c5a685f7cf7d99a0ca6b84524d Mon Sep 17 00:00:00 2001
From: Chenyan Feng <ella.feng@nxp.com>
Date: Sat, 24 Jun 2017 16:03:44 +0800
Subject: [PATCH 3973/5242] MGS-3024 dts: update i.MX8MQ device config to
 enable GPU

commit  2695b108130f5b48cdcecb728d9f56ed792aab8f from
https://source.codeaurora.org/external/imx/linux-imx.git

enable gpu device in imx8mq-evk board,
increase GPU memory size from 32M to 128M,
enable GPU flat mappping for full DDR range.

add IMX8MQ_CLK_GPU_AHB_DIV into gpu clock list.

Signed-of-by: Chenyan Feng <ella.feng@nxp.com>

This was skipped during the rebase but now that GPU is enabled we can
also enable it on 8mq.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |    4 ++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi    |    9 +++++----
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 9049893..4381165 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -760,6 +760,10 @@
 	power-supply = <&sw1c_reg>;
 };
 
+&gpu {
+	status = "okay";
+};
+
 &vpu {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index f371222..38d102f 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -1181,16 +1181,17 @@
 
 	gpu: gpu@38000000 {
 		compatible = "fsl,imx8mq-gpu", "fsl,imx6q-gpu";
-		reg = <0x0 0x38000000 0 0x40000>, <0x0 0x60000000 0x0 0x14000000>, <0x0 0x0 0x0 0x2000000>;
+		reg = <0x0 0x38000000 0 0x40000>, <0x0 0x40000000 0x0 0xC0000000>, <0x0 0x0 0x0 0x8000000>;
 		reg-names = "iobase_3d", "phys_baseaddr", "contiguous_mem";
-		interrupts = <0 3 0x4>;
+		interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "irq_3d";
-		clocks = <&clk IMX8MQ_CLK_GPU_ROOT>, <&clk IMX8MQ_CLK_GPU_SHADER_DIV>, <&clk IMX8MQ_CLK_GPU_AXI_DIV>;
-		clock-names = "gpu3d_clk", "gpu3d_shader_clk", "gpu3d_axi_clk";
+		clocks = <&clk IMX8MQ_CLK_GPU_ROOT>, <&clk IMX8MQ_CLK_GPU_SHADER_DIV>, <&clk IMX8MQ_CLK_GPU_AXI_DIV>, <&clk IMX8MQ_CLK_GPU_AHB_DIV>;
+		clock-names = "gpu3d_clk", "gpu3d_shader_clk", "gpu3d_axi_clk", "gpu3d_ahb_clk";
 		assigned-clocks = <&clk IMX8MQ_CLK_GPU_CORE_SRC>, <&clk IMX8MQ_CLK_GPU_SHADER_SRC>, <&clk IMX8MQ_CLK_GPU_AXI_SRC>, <&clk IMX8MQ_CLK_GPU_AHB_SRC>;
 		assigned-clock-parents = <&clk IMX8MQ_GPU_PLL_OUT>, <&clk IMX8MQ_GPU_PLL_OUT>, <&clk IMX8MQ_GPU_PLL_OUT>, <&clk IMX8MQ_GPU_PLL_OUT>;
 		assigned-clock-rates = <800000000>, <800000000>, <800000000>, <800000000>;
 		power-domains = <&gpu_pd>;
+		depth-compression = <0>;
 		status = "disabled";
 	};
 
-- 
1.7.9.5

